### DISCUSSION TITLE
kbe 新手编译部署问题总结

### DISCUSSION THREAD

**Discussion by ddonano** [kbengine/kbengine#1586] [General]
1、查看gcc g++ openssl版本号 
gcc --version
g++ --version
openssl version

2、比较合适的版本是 gcc g++ < 7, openssl =  1.1
3、 安装依赖包：
ubuntu: 
sudo apt install g++ build-essential curl  libtool autoconf libssl-dev mysql-server libmysqlclient-dev
centos:
sudo yum groupinstall "Development Tools"
sudo  yum install openssl-devel libtool autoconf mysql-server mysql-devel

4、编译 到kbengine\kbe\src目录，使用make命令：
 （1）gcc高于7 需要配置：
#kbe/src/build/common.mak第286行增加
CPPFLAGS += -Wno-format-truncation

（2）在Centos8下,我们编译Kbengine会出现找不到rpc.h头文件的情况：

具体报错如下
sigar_util.c: 在文件作用域：
sigar_util.c:742:10: 致命错误：rpc/rpc.h：没有那个文件或目录
#include <rpc/rpc.h>
          ^~~~~~~~~~~
这是因为,在Centos8中使用了新库libtirpc。
安装libtirpc
sudo yum -y install libtirpc-devel

5、#mysql8.x创建用户和赋予权限，需要分开
#创建用户
create user 'kbe'@'%' identified by 'pwd123456';
#授权
grant all privileges on *.* to 'kbe'@'%';

6、centos 8 编译了mysql 8.0的客户端导致服务启动后，dbmgr 没连上，无法执行sql，我的sql 服务在另外一台机器上，是5.7版本的：
回退 mysql client dev 到 5.7版本，解决问题

目前仍然有两个问题：
1、自己本地wsl 遇到多网卡问题，导致启动不起来，暂未解决，后来 放弃使用wsl，使用vs 20019 windows启动方式
2、集群部署方式不知道咋部署的

**Comment by Wandar** [comment]
感谢分享
多网卡试试/sbin/ip route add broadcast 255.255.255.255 dev eth1
